[docs][windows]: Pod OS field update (#30436)
kubernetes/website: content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md

keep Linux and Windows workloads on their respective OS-specific nodes. 
This likely imposes a burden only on Windows users. The recommended approach is outlined below, 
with one of its main goals being that this approach should not break compatibility for existing Linux workloads.

>  {{< note >}}

Author: @jsturtevant - I added a note here. PTAL.

>  If the `IdentifyPodOS` featuregate is enabled and the `pod.spec.os.name` is set to `windows` it will have

Reviewer: ```suggestion
If the `IdentifyPodOS` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is
enabled, you can (and should) set `.spec.os.name` for a Pod to indicate the operating system
that the containers in that Pod are designed for. For Pods that run Linux containers, set
`.spec.os.name` to `linux`. For Pods that run Windows containers, set `.spec.os.name`
to Windows.

The scheduler does not use the value of `.spec.os.name` when assigning Pods to nodes. You should
use normal Kubernetes mechanisms for
[assigning pods to nodes](/docs/concepts/scheduling-eviction/assign-pod-node/)
to ensure that the control plane for your cluster places pods onto nodes that are running the
appropriate operating system.
```

>  If the `IdentifyPodOS` featuregate is enabled and the `pod.spec.os.name` field even if set to `windows` will have

Reviewer 2: ```suggestion
 If the `IdentifyPodOS` featuregate is enabled and the `pod.spec.os.name` is set to `windows` it will have
```

>  no effect on the scheduling of the Windows pods, so taints and tolerations and node selectors are still required.

Reviewer 2: ```suggestion
 no effect on the scheduling of the Windows pods, so taints and tolerations and node selectors are still required
```

>  no effect on the scheduling of the Windows pods, so we should still rely on taints and tolerations and node selector

Reviewer 2: ```suggestion
 no effect on the scheduling of the Windows pods, so taints and tolerations and node selectors are still required.
```

Maybe link to that section?

Reviewer 2: I understand we can't point to the future in the docs but for my own understanding, once we are in Beta we should be able to use this field for scheduling?

Author: We've not decided on it yet. The main reason for not going in that direction atm is LCOW(where the nodeSelector may point to Windows but the podOS can be linux).
